### ISSUE TITLE
Run Script for Amplify tools can't find node when nvm is used

### ISSUE THREAD

**Issue by attilah** [aws-amplify/amplify-ios#488]
**Describe the bug**
Following the Todo guide when I try to build the project, the Run Script phase fails with the error:

```
env: node: Not a directory
warning: Node is not installed. Vist https://nodejs.org/en/download/ to install it
Command PhaseScriptExecution failed with a nonzero exit code
```

I'm using node through nvm and does work from `sh`, `bash`, `zsh`, but not from xcode script.

I even tried to add this to the script, but did not help:
```
export PATH="$PATH:/Users/<user>/.nvm/versions/node/v13.12.0/bin/node"
```

**Comment by drochetti** [comment]
A temporary mitigation for this issue is to prepend to your Amplify Build Phase Script the NVM initialization: `. ~/.nvm/nvm.sh;`

A complete example of the Build Phase script:

```sh
. ~/.nvm/nvm.sh;
${PODS_ROOT}/AmplifyTools/amplify-tools.sh;
```

**Comment by drochetti** [comment]
This was fixed as part of [1.0.2](https://github.com/aws-amplify/amplify-ios/releases/tag/v1.0.2). Please re-open if you're still experiencing the same issue.

**Comment by roxberry** [comment]
Started on Xcode 12 beta 6, adding . ~/.nvm/nvm.sh; resolves it. Using amplify 4.29.1
